### DISCUSSION TITLE
The unit of S and C? how to set a threshold on spikes?

### DISCUSSION THREAD

**Discussion by Yifang-Yuan** [miniscope/minian#139] [Q&A]
Hi there,
1. I noticed that the units of transient and spikes we get on the visualized plots are A.U., which is quite confusing, how can we convert it to dF/F, or some more reasonable units for firing rate?
2. The spike value I got range from 10^-2 to 10^2, the algorithm gives all non-zero spikes, do you have any suggestions to set a threshold to get the real spikes? For example shall I use a mean, or some absolute value like 10?  The point is, the absolute value of the spike might be different for every session/cell/behavior state, so using a mean may not be the best choice.

**Comment by phildong** [comment]
1. I'm not sure what's the usual unit for dF/F. I don't think there is a deterministic relationship between the values of C/S with the firing rate. They are basically whatever numerical values that fit the fluorescent values (which are quite arbitrary by themselves).
2. you can use some adaptive thresholding algorithm. For example, [Gaussian mixture model](https://scikit-learn.org/stable/modules/mixture.html) with two components, or maybe some [histogram-based thresholding](https://scikit-image.org/docs/0.13.x/auto_examples/xx_applications/plot_thresholding.html) would work.

**Comment by ZachPenn** [comment]
As @phildong said, binarizing spikes in some way can be helpful so that you can then get spikes/unit time.  I've tried both OTSU thresholding and gaussian mixture models to binarize spikes from each cell, and GMMs appeared to work a bit better (just eye-balling).  GMMs are very simple to fit using scikit-learn.  The only issue with binarizing data in this way is that you lose some info that may be conveyed in large vs low spikes.

**Reply by Yifang-Yuan** [reply]
Thanks, will try!
